packages/examples/src/langium/config/wrapperStatemachineConfig.ts

```diff
@@ -17,56 +25,52 @@ export const createLangiumGlobalConfig = async (htmlElement: HTMLElement): Promi
-                enableKeybindingsService: true,
                 debugLogging: true
             },
             editorAppConfig: {
                 $type: 'vscodeApi',
                 languageId: 'statemachine',
                 code: code,
                 useDiffEditor: false,
-                extension: {
-                    name: 'langium-example',
-                    publisher: 'monaco-languageclient-project',
-                    version: '1.0.0',
-                    engines: {
-                        vscode: '*'
+                // Ensure all required extensions are loaded before setting up the language extension
+                awaitExtensionReadiness: [whenReadyThemes],
+                extensions: [{
+                    config: {
+                        name: 'statemachine-example',
+                        publisher: 'monaco-editor-wrapper-examples',
+                        version: '1.0.0',
+                        engines: {
+                            vscode: '*'
+                        },
+                        contributes: {
+                            languages: [{
+                                id: 'statemachine',
+                                extensions: ['.statemachine'],
+                                aliases: ['statemachine', 'Statemachine'],
+                                configuration: './statemachine-configuration.json'
+                            }],
+                            grammars: [{
+                                language: 'statemachine',
+                                scopeName: 'source.statemachine',
+                                path: './statemachine-grammar.json'
+                            }]
+                        }
                     },
-                    contributes: {
-                        languages: [{
-                            id: 'statemachine',
-                            extensions: [
-                                '.statemachine'
-                            ],
-                            aliases: [
-                                'statemachine',
-                                'Statemachine'
-                            ],
-                            configuration: './statemachine-configuration.json'
-                        }],
-                        grammars: [{
-                            language: 'statemachine',
-                            scopeName: 'source.statemachine',
-                            path: './statemachine-grammar.json'
-                        }]
-                    }
-                },
-                extensionFilesOrContents: extensionFilesOrContents,
+                    filesOrContents: extensionFilesOrContents
+                }],
                 userConfiguration: {
-                    json: `{
-    "workbench.colorTheme": "Default Dark Modern",
-    "editor.guides.bracketPairsHorizontal": "active",
-    "editor.lightbulb.enabled": true
-}`
+                    json: JSON.stringify({
```

Comment:
This feels better than the raw JSON text from before.

Reply:
I thought so, too.